examples/acoustic/gradient_example.py

```diff
@@ -0,0 +1,176 @@
+                       coordinates=receiver_coords)
+    # Receiver for smoothed model
+    rec_s = SourceLike(name="rec_s", dimensions=[time, p_rec], npoint=nrec, nt=nt, dt=dt,
+                       h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                       coordinates=receiver_coords)
+
+    # Create the forward wavefield to use (only 3 timesteps)
+    # Once checkpointing is in, this will be the only wavefield we need
+    u_nosave = TimeData(name="u_ns", shape=model.shape_domain, time_dim=nt,
+                        time_order=time_order, space_order=space_order, save=False,
+                        dtype=model.dtype)
+
+    # Forward wavefield where all timesteps are stored
+    # With checkpointing this should go away <----
+    u_save = TimeData(name="u_s", shape=model.shape_domain, time_dim=nt,
+                      time_order=time_order, space_order=space_order, save=True,
+                      dtype=model.dtype)
+
+    # Forward Operators - one with save = True and one with save = False
+    fw = ForwardOperator(model, u_save, src, rec_t, time_order=time_order,
+                         spc_order=space_order, save=True, cache_blocking=cache_blocking,
+                         dse=dse, dle=dle, compiler=compiler, profile=True)
+
+    fw_nosave = ForwardOperator(model, u_nosave, src, rec_t, time_order=time_order,
+                                spc_order=space_order, save=False,
+                                cache_blocking=cache_blocking, dse=dse, dle=dle,
+                                compiler=compiler, profile=True)
+
+    # Calculate receiver data for true velocity
+    fw_nosave.apply(autotuning=autotuning, rec_t=rec_t)
+
+    # Change to the smooth velocity
+    model.m.data[:] = model.pad(1 / initial_vp ** 2)
+
+    # Smooth velocity
+    # This is the pass that needs checkpointing <----
+    fw.apply(autotuning=autotuning, rec_t=rec_s)
+
+    # Objective function value
+    F0 = .5*linalg.norm(rec_s.data - rec_t.data)**2
+
+    # Receiver for Gradient
+    # Confusing nomenclature because this is actually the source for the adjoint
+    # mode
+    rec_g = SourceLike(name="rec_g", dimensions=[time, p_rec], npoint=nrec, nt=nt, dt=dt,
+                       h=h, ndim=ndim, nbpml=nbpml, dtype=dtype,
+                       coordinates=receiver_coords)
+    rec_g.data[:] = rec_s.data - rec_t.data
+
+    # Gradient symbol
+    grad = DenseData(name="grad", shape=model.shape_domain, dtype=model.dtype)
+    # Reusing u_nosave from above as the adjoint wavefield since it is a temp var anyway
+    gradop = GradientOperator(model, u_nosave, grad, rec_g, u_save, time_order=time_order,
+                              spc_order=space_order, cache_blocking=cache_blocking,
+                              dse=dse, dle=dle, compiler=compiler, profile=True)
+
+    # Clear the wavefield variable to reuse it
+    # This time it represents the adjoint field
+    u_nosave.data.fill(0)
```

Comment:
I take it `.data.fill` is a custom Devito function? Why not stick to native Python instead `u_nosave.data[:] = 0.`?